src/type-coercions.md

```diff
@@ -184,3 +184,48 @@ unsized coercion to `Foo<U>`.
 
 [`Unsize`]: ../std/marker/trait.Unsize.html
 [`CoerceUnsized`]: ../std/ops/trait.CoerceUnsized.html
+
+## LUB Coercion
+
+This is not in the RFC 401, but it's the behavior of current compiler. In LUB
+Coercion, we tried to find the least upper bound of given types. It can be
+observed in array collection, if-else/match arms.
+
+For example:
+
+```rust
```

Comment:
I'd like to leave the variables here since I'll add a example where these variables are annoted with types. But I changed the variable names to the classic combination `foo`, `bar` and `baz`.